hcn/hcn.go

```diff
@@ -0,0 +1,142 @@
+//sys hcnDeleteNamespace(id *guid.GUID, result **uint16) (hr error) = computenetwork.HcnDeleteNamespace?
+//sys hcnCloseNamespace(namespace hcnNamespace) (hr error) = computenetwork.HcnCloseNamespace?
+
+// LoadBalancer
+//sys hcnEnumerateLoadBalancers(query string, loadBalancers **uint16, result **uint16) (hr error) = computenetwork.HcnEnumerateLoadBalancers?
+//sys hcnCreateLoadBalancer(id *guid.GUID, settings string, loadBalancer *hcnLoadBalancer, result **uint16) (hr error) = computenetwork.HcnCreateLoadBalancer?
+//sys hcnOpenLoadBalancer(id *guid.GUID, loadBalancer *hcnLoadBalancer, result **uint16) (hr error) = computenetwork.HcnOpenLoadBalancer?
+//sys hcnModifyLoadBalancer(loadBalancer hcnLoadBalancer, settings string, result **uint16) (hr error) = computenetwork.HcnModifyLoadBalancer?
+//sys hcnQueryLoadBalancerProperties(loadBalancer hcnLoadBalancer, query string, properties **uint16, result **uint16) (hr error) = computenetwork.HcnQueryLoadBalancerProperties?
+//sys hcnDeleteLoadBalancer(id *guid.GUID, result **uint16) (hr error) = computenetwork.HcnDeleteLoadBalancer?
+//sys hcnCloseLoadBalancer(loadBalancer hcnLoadBalancer) (hr error) = computenetwork.HcnCloseLoadBalancer?
+
+// Service
+//sys hcnOpenService(service *hcnService, result **uint16) (hr error) = computenetwork.HcnOpenService?
+//sys hcnRegisterServiceCallback(service hcnService, callback int32, context int32, callbackHandle *hcnCallbackHandle) (hr error) = computenetwork.HcnRegisterServiceCallback?
+//sys hcnUnregisterServiceCallback(callbackHandle hcnCallbackHandle) (hr error) = computenetwork.HcnUnregisterServiceCallback?
+//sys hcnCloseService(service hcnService) (hr error) = computenetwork.HcnCloseService?
+
+type hcnNetwork syscall.Handle
+type hcnEndpoint syscall.Handle
+type hcnNamespace syscall.Handle
+type hcnLoadBalancer syscall.Handle
+type hcnService syscall.Handle
+type hcnCallbackHandle syscall.Handle
+
+// CheckForErrors looks for non-zero HRESULT and looks at the result buffer.
+func CheckForErrors(methodName string, hr error, resultBuffer *uint16) error {
+	errorFound := false
+
+	if hr != nil {
+		errorFound = true
+	}
+
+	result := ""
+	if resultBuffer != nil {
+		result = interop.ConvertAndFreeCoTaskMemString(resultBuffer)
+		if result != "" {
+			errorFound = true
+		}
+	}
+
+	if errorFound {
+		return hcserror.New(hr, methodName, result)
+	}
+
+	return nil
+}
+
+// SchemaVersion for HCN Objects/Queries.
+type SchemaVersion struct {
+	Major uint32 `json:",omitempty"`
+	Minor uint32 `json:",omitempty"`
+}
+
+// MarshalJSON formats the SchemaVersion.
+func (s *SchemaVersion) MarshalJSON() ([]byte, error) {
+	jsonString := fmt.Sprintf("{\"Major\":%d,\"Minor\":%d}", s.Major, s.Minor)
+	return []byte(jsonString), nil
+}
+
+// HostComputeQuery is the format for HCN queries.
+type HostComputeQuery struct {
+	SchemaVersion SchemaVersion `json:""`
+	Flags         uint32        `json:",omitempty"` // 0: None, 1: Detailed
```

Comment:
Define:
```go
type HostComputeQueryFlags uint32

var (
    HostComputeQueryFlagsNone HostComputeQueryFlags = 1
    HostComputeQueryFlagsDetailed = 1
)
```
Patterns like this really help people to know what is possible via the type system instead of a comment.